### PR TITLE
adding app-level signal for dirty state changes

### PR DESCRIPTION
**Pull request by dharmaquark**
my proposed change for #4840

### CODE REVIEW

**Comment by dharmaquark** [comment]
Awesome, thanks!

File: packages/application/src/index.ts [hunk 1]
```diff
@@ -105,6 +106,13 @@ export class JupyterLab extends Application<ApplicationShell> {
     return this._busySignal;
   }
 
+  /**
+   * Returns a signal for when application changes it's dirty status.
```
Comment (jasongrout) [inline packages/application/src/index.ts]: its, not it's

Comment (jasongrout) [reply packages/application/src/index.ts]: (looks like busySignal also has the same grammar issue)

**Comment by jasongrout** [comment]
Last changes are just doc changes, and previous tests passed, so merging. Thanks!

**Pull request merged**